drivers/led/is31fl3216a.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2023 Endor AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT issi_is31fl3216a
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/logging/log.h>
+
+#define IS31FL3216A_REG_CONFIG		0x00
+#define IS31FL3216A_REG_CTL_1		0x01
+#define IS31FL3216A_REG_CTL_2		0x02
+#define IS31FL3216A_REG_LIGHT_EFFECT	0x03
+#define IS31FL3216A_REG_CHANNEL_CONFIG	0x04
+#define IS31FL3216A_REG_GPIO_CONFIG	0x05
+#define IS31FL3216A_REG_OUTPUT_PORT	0x06
+#define IS31FL3216A_REG_INT_CONTROL	0x07
+#define IS31FL3216A_REG_ADC_SAMPLE_RATE	0x09
+#define IS31FL3216A_REG_PWM_FIRST	0x10
+#define IS31FL3216A_REG_PWM_LAST	0x1F
+#define IS31FL3216A_REG_UPDATE		0xB0
+#define IS31FL3216A_REG_FRAME_DELAY	0xB6
+#define IS31FL3216A_REG_FRAME_START	0xB7
+
+#define IS31FL3216A_MAX_LEDS		16
+
+LOG_MODULE_REGISTER(is31fl3216a, CONFIG_LED_LOG_LEVEL);
+
+struct is31fl3216a_cfg {
+	struct i2c_dt_spec i2c;
+};
+
+static int is31fl3216a_write_buffer(const struct i2c_dt_spec *i2c,
+				    const uint8_t *buffer, uint32_t num_bytes)
+{
+	int status;
+
+	status = i2c_write_dt(i2c, buffer, num_bytes);
+	if (status) {
```

Comment:
Could you replace that with `if (ret < 0)`  (or `if (status < 0)` if you want to keep `status` variable). everywhere?

Reply:
Yes